File: src/main/java/org/devignite/reportBuilder/dataProvider/MongoDataProviderSettings.java
package org.devignite.reportBuilder.dataProvider;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.devignite.reportBuilder.dataProvider.abstractions.DataProviderSettings;

@Data
@EqualsAndHashCode(callSuper = false)
public class MongoDataProviderSettings extends DataProviderSettings {
    private String sourceCollectionName;
}

File: dotnet/library/DataProvider/Mongo/MongoDataProviderSettings.cs
namespace DevIgnite.ReportBuilderLibrary.DataProvider {
    public interface IDataProviderSettings { }

    public class MongoDataProviderSettings : IDataProviderSettings {
        public MongoDataProviderSettings(string databaseName, string collectionName) {
            DatabaseName = databaseName;
            CollectionName = collectionName;
        }

        public string DatabaseName { get; }
        public string CollectionName { get; }
    }
}